src/test/kotlin/no/nav/syfo/identhendelse/IdenthendelseServiceSpek.kt

```diff
@@ -0,0 +1,154 @@
+package no.nav.syfo.identhendelse
+
+import io.ktor.server.testing.*
+import kotlinx.coroutines.*
+import no.nav.syfo.application.cache.RedisStore
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.domain.PersonIdent
+import no.nav.syfo.personstatus.db.createPersonOversiktStatus
+import no.nav.syfo.personstatus.db.getPersonOversiktStatusList
+import no.nav.syfo.personstatus.domain.PersonOversiktStatus
+import no.nav.syfo.testutil.ExternalMockEnvironment
+import no.nav.syfo.testutil.UserConstants
+import no.nav.syfo.testutil.dropData
+import no.nav.syfo.testutil.generator.generateKafkaIdenthendelseDTOGenerator
+import org.amshove.kluent.internal.assertFailsWith
+import org.amshove.kluent.shouldBeEqualTo
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object IdenthendelseServiceSpek : Spek({
+
+    describe(IdenthendelseServiceSpek::class.java.simpleName) {
+
+        with(TestApplicationEngine()) {
+            start()
+
+            val externalMockEnvironment = ExternalMockEnvironment.instance
+            val database = externalMockEnvironment.database
+            val pdlClient = PdlClient(
+                azureAdClient = AzureAdClient(
+                    azureEnvironment = externalMockEnvironment.environment.azure,
+                    redisStore = RedisStore(externalMockEnvironment.environment.redis),
+                ),
+                clientEnvironment = externalMockEnvironment.environment.clients.pdl,
+                redisStore = RedisStore(externalMockEnvironment.environment.redis),
+            )
+
+            val identhendelseService = IdenthendelseService(
+                database = database,
+                pdlClient = pdlClient,
+            )
+
+            beforeEachTest {
+                database.connection.dropData()
+            }
+
+            describe("Happy path") {
```

Comment:
Trenger man kommentarer overalt her?

Reply:
Nei egentlig ikke, det er kanskje mest for min egen del? Jeg føler det er vanskelig å lese tester noen ganger fordi man ofte prøver å reprodusere visse usecaser, og da tenkte jeg det var lettere å forstår _hvorfor_ man gjør de tingene man gjør i testen (i tillegg til _hva_ som skjer, som koden i seg selv dokumenterer).